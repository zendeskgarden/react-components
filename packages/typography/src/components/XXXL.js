/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DEFAULT_THEME, retrieveComponentStyles, isRtl } from '@zendeskgarden/react-theming';

const COMPONENT_ID = 'typography.xxxl';

const StyledXXXL = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION
})`
  line-height: ${props => props.theme.lineHeights.xxxl};
  font-size: ${props => props.theme.fontSizes.xxxl};

  direction: ${props => (isRtl(props) ? 'rtl' : 'ltr')};

  ${props => retrieveComponentStyles(COMPONENT_ID, props)};
`;

StyledXXXL.defaultProps = {
  theme: DEFAULT_THEME
};

/**
 * Accepts all standard props relating to provided `tag`
 */
const XXXL = ({ tag, ...other }) => {
  const CustomTagXXXL = StyledXXXL.withComponent(tag);

  return <CustomTagXXXL {...other} />;
};

XXXL.propTypes = {
  /** Any valid DOM element for the styled component */
  tag: PropTypes.any
};

XXXL.defaultProps = {
  tag: 'div'
};

/** @component */
export default XXXL;
