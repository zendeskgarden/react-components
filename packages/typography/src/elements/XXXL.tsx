/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { StyledFont } from '../styled';
import { ITypescaleProps } from '../types';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const XXXL = forwardRef<HTMLDivElement, ITypescaleProps>(
  ({ isBold, tag, ...other }, ref) => (
    <StyledFont $isBold={isBold} $size="3xlarge" {...other} as={tag} ref={ref} />
  )
);

XXXL.displayName = 'XXXL';

XXXL.propTypes = {
  tag: PropTypes.any,
  isBold: PropTypes.bool
};

XXXL.defaultProps = {
  tag: 'div'
};
