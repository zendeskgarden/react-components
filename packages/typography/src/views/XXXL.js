/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { zdFontSizeXxxl, zdLineHeightXxxl } from '@zendeskgarden/css-variables';
import { retrieveTheme, isRtl } from '@zendeskgarden/react-theming';

const COMPONENT_ID = 'typography.xxxl';

const StyledXXXL = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION
})`
  line-height: ${zdLineHeightXxxl};
  font-size: ${zdFontSizeXxxl};

  direction: ${props => (isRtl(props) ? 'rtl' : 'ltr')};

  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/**
 * Accepts all standard props relating to provided `tag`
 */
const XXXL = ({ tag, ...other }) => {
  const CustomTagXXXL = StyledXXXL.withComponent(tag);

  return <CustomTagXXXL {...other} />;
};

XXXL.propTypes = {
  /** Any valid DOM element for the styled component */
  tag: PropTypes.string
};

XXXL.defaultProps = {
  tag: 'div'
};

/** @component */
export default XXXL;
