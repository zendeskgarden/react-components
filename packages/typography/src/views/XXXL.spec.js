/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { renderRtl } from 'garden-test-utils';
import XXXL from './XXXL';

describe('XXXL', () => {
  const Example = props => <XXXL {...props}>Hello world</XXXL>;

  it('applies correct styling with RTL locale', () => {
    const { container } = renderRtl(<Example />);

    expect(container.firstChild).toHaveStyleRule('direction', 'rtl');
  });
});
