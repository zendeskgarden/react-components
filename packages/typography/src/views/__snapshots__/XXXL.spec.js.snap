// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XXXL applies correct styling by default 1`] = `
.c0 {
  line-height: zdLineHeightXxxl;
  font-size: zdFontSizeXxxl;
  direction: ltr;
}

<Example>
  <XXXL
    tag="div"
  >
    <XXXL__StyledXXXL>
      <div
        className="-div c0"
        data-garden-id="typography.xxxl"
        data-garden-version="version"
      >
        Hello world
      </div>
    </XXXL__StyledXXXL>
  </XXXL>
</Example>
`;

exports[`XXXL applies correct styling with RTL locale 1`] = `
.c0 {
  line-height: zdLineHeightXxxl;
  font-size: zdFontSizeXxxl;
  direction: rtl;
}

<Example>
  <XXXL
    tag="div"
  >
    <XXXL__StyledXXXL>
      <div
        className="-div c0"
        data-garden-id="typography.xxxl"
        data-garden-version="version"
      >
        Hello world
      </div>
    </XXXL__StyledXXXL>
  </XXXL>
</Example>
`;
